src/handlers/shared/start.ts

```diff
@@ -37,21 +38,19 @@ export async function start(context: Context, issue: Context["payload"]["issue"]
   // check max assigned issues
 
   const openedPullRequests = await getAvailableOpenedPullRequests(context, sender.login);
-  logger.info(`Opened Pull Requests with approved reviews or with no reviews but over 24 hours have passed: ${JSON.stringify(openedPullRequests)}`);
+  logger.info(`Opened Pull Requests with approved reviews or with no reviews but over 24 hours have passed`, {
+    openedPullRequests: openedPullRequests.map((pr) => pr.html_url),
+  });
 
   const assignedIssues = await getAssignedIssues(context, sender.login);
-  logger.info("Max issue allowed is", { maxConcurrentTasks });
+  logger.info("Max issues allowed is", { limit: maxTask.limit, assigned: assignedIssues.length });
 
   // check for max and enforce max
 
-  if (assignedIssues.length - openedPullRequests.length >= maxConcurrentTasks) {
-    const log = logger.error("Too many assigned issues, you have reached your max limit", {
-      assignedIssues: assignedIssues.length,
-      openedPullRequests: openedPullRequests.length,
-      maxConcurrentTasks,
-    });
+  if (Math.abs(assignedIssues.length - openedPullRequests.length) >= maxTask.limit) {
```

Comment:
Wouldn't this mean that if you have no assigned issues but plenty of open pull-request, you would get limited even if you actually are assigned to nothing?

Reply:
You raise a valid concern, but isn’t this approach actually preferable? Without this check, someone could bypass the task limit simply by opening multiple PRs. Essentially, this would make the limit meaningless.

Consider this scenario:

`maxLimit = 5`
`assignedIssues = 0`
`openPRs = 6`

The calculation would be `0 - 6 == -6` > `-6 + 5 == -1` > `-1 >= 5`. And for every new task, a new PR which means unlimited tasks.

So, with each additional PR, the max limit would increase, defeating the purpose of having a limit at all. This check ensures that the task limit is enforced properly, preventing such loopholes.

I could imagine a situation where a user starts a task, then opens a PR with some "filler" commits just to hold their place, or for some other strategic reason. This check helps prevent that kind of workaround.

Reply:
I also agree with you. I guess any pull-request should have a related issue specification, I was thinking more when we open a bunch of PRs that actually are quick fixes. But I guess it is better to secure it that way.

Reply:
You mean from a core team perspective I figure? In this case we could 

- bypass the check all together
- have org member checks configurable
- factor it into the config settings when defining `MEMBER` and have it be like 20-odds which seems reasonable
- ...

Reply:
With how the review process looks like here, that will make things really rough.

In a hypothetical scenario where a contributor has a limit of 3 task per time, successfully gets the tasks done (including test and everything related ) and the review process starts. From how the speed of the whole review process looks like here, it could take months to close PR's. It can get frustrating very fast. 

In the V1 bot, I thought this was a feature intentionally introduced to allow for continuation of work during the review process. Didn't know it was just a loophole we forgot to cover 😅

Reply:
In my experience, contributor's within the first 3-5 months likely cannot efficiently manage more than two or three concurrent tasks anyway. After this code review moves a lot faster and it is no longer such an issue for them, at least was the case for me and all I've seen this I started

> In the V1 bot, I thought this was a feature intentionally introduced to allow for continuation of work during the review process. Didn't know it was just a loophole we forgot to cover 

It was an intentional feature but it's being improved vis-a-vis `automated-merging`, review incentives etc. I'd personally rather see a contributor take a PR from 0 - 100 before moving on to the next task.

Reply:
I didn't scrutinize the code because I'm mobile but I can tell a little story about v1

---

We had this problem where contributors would self assign a ton of tasks and then block progress. So we made a limit of how many open tasks can be actively assigned. 

Then we realized that our review team was actually causing the delays, so we decided that after a 24 hour grace period of pending review, we would lift the limit by one. 

However if a reviewer requested changes on any pull, then the limit is reduced again, which forces the contributor to focus on the open pulls instead of starting new tasks. 

At a high level, we realized that delays can happen on both the contributor and reviewer side. This system seemed to work quite well in old v1, and we should reimplement.